File: testapp/src/main/java/com/nerdscorner/mvplib/testapp/interfaces/attribute/presenter/AttributeMainPresenter.java
package com.nerdscorner.mvplib.testapp.interfaces.attribute.presenter;

import com.nerdscorner.mvplib.interfaces.presenter.BaseActivityPresenter;
import com.nerdscorner.mvplib.testapp.interfaces.attribute.model.AttributeMainModel;
import com.nerdscorner.mvplib.testapp.interfaces.attribute.model.AttributeMainModelInterface;
import com.nerdscorner.mvplib.testapp.interfaces.attribute.view.AttributeMainView;
import com.nerdscorner.mvplib.testapp.interfaces.attribute.view.AttributeMainViewInterface;

public class AttributeMainPresenter extends BaseActivityPresenter<AttributeMainViewInterface, AttributeMainModelInterface> implements AttributeMainPresenterInterface {

    public AttributeMainPresenter(AttributeMainView view, AttributeMainModel model) {
        super(view, model);
    }

    @Override
    public void onActionClicked() {
        view.setTextValue("Executing background task...");
        model.doSomethingInBackground();
    }

    @Override
    public void onBackgroundTaskCompleted() {
        view.setTextValue("Background task completed");
    }
}

File: testapp/src/main/java/com/github/marcherdiego/mvp/testapp/ui/mvp/presenter/AttributeMainPresenter.kt
package com.github.marcherdiego.mvp.testapp.ui.mvp.presenter

import com.github.marcherdiego.mvp.events.bus.Bus
import com.github.marcherdiego.mvp.events.presenter.BaseActivityPresenter
import com.github.marcherdiego.mvp.testapp.ui.mvp.model.AttributeMainModel
import com.github.marcherdiego.mvp.testapp.ui.mvp.model.AttributeMainModel.BackgroundTaskCompletedEvent
import com.github.marcherdiego.mvp.testapp.ui.mvp.view.AttributeMainView
import com.github.marcherdiego.mvp.testapp.ui.mvp.view.AttributeMainView.ActionClickedEvent
import org.greenrobot.eventbus.Subscribe

class AttributeMainPresenter(view: AttributeMainView, model: AttributeMainModel, bus: Bus) :
    BaseActivityPresenter<AttributeMainView, AttributeMainModel>(view, model, bus) {

    @Subscribe
    fun onActionClicked(event: ActionClickedEvent) {
        view.setTextValue("Executing background task...")
        model.doSomethingInBackground()
    }

    @Subscribe
    fun onBackgroundTaskCompleted(event: BackgroundTaskCompletedEvent) {
        view.setTextValue("Background task completed ${event.data}")
    }

    override fun onResume() {
        super.onResume()
        view.withActivity {
        }
    }
}
